### PR TITLE
add examples

### PR DESCRIPTION
**Pull request by codehag**
Hopefully addresses any ambiguities such as those raised by #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +65,15 @@ Type I is used by user libraries, as well as by the web platform in WebIDL and D
 
 Type II is supported if built-in methods create new instances of the subclass. For example, if `Array.prototype.map` or `Array.from` returns instances of subclasses of `Array`. Support for this is provided by delegating to `this.constructor[`@@species`]` inside built-in methods for the default @@species getter.
```
Comment (littledan) [inline README.md:66]: Should `@@species` be mentioned here? What is meant by it?

File: README.md [hunk 2]
```diff
@@ -81,6 +113,21 @@ Type IV is supported if built-in methods consult properties on instances instead
 
 Note that `RegExp`'s @@match, @@matchAll, @@replace, @@search, and @@split symbols themselves aren't strictly only for subclassing support, as they are not used in `RegExp` methods themselves. Instead, they are used as a protocol for `String` so that completely custom `RegExp` instances may be consumed.
 
+### Example
+
+```js
+class R extends RegExp {
```
Comment (domenic) [inline README.md:119]: This example should probably contain an empty constructor that does not call `super()`.

File: README.md [hunk 3]
```diff
@@ -43,6 +43,18 @@ Supporting built-in subclassing has also negatively affected authoring of new bu
...
 
+### Example
+
+```js
+class A extends Array {
+  constructor() {
+    super();
+  }
+}
+A.from([1,2,3])     // return type: Array
```
Comment (syg) [inline README.md:54]: The example should probably be more like `new A // return type: A` to show that `new.target` is propagated and determines the [[Prototype]] of the created instance.

File: README.md (hunk 2 above)
Comment (syg) [reply README.md:119]: +1

File: README.md [hunk 4]
```diff
@@ -81,6 +113,21 @@ Type IV is supported if built-in methods consult properties on instances instead
 
 Note that `RegExp`'s @@match, @@matchAll, @@replace, @@search, and @@split symbols themselves aren't strictly only for subclassing support, as they are not used in `RegExp` methods themselves. Instead, they are used as a protocol for `String` so that completely custom `RegExp` instances may be consumed.
 
+### Example
+
+```js
+class R extends RegExp {
+  [Symbol.match](str) {
```
Comment (syg) [inline README.md:120]: I'd remove overriding of `@@match`. The overridden `exec` already gets called by the default implementation of `@@match`, which is the bad behavior that V8 is concerned with.

File: README.md [hunk 5]
```diff
@@ -181,6 +264,31 @@ The following methods on `String.prototype` will be modified to check for [[RegE
...
+After this change:
+
+```js
+class R extends RegExp {
+  [Symbol.match](str) {
+    return "overridden";
+  }
+}
+console.log("some string".match(new R("foo")))     // null
+```
```
Comment (syg) [inline README.md:290]: This example doesn't capture the intent. Sorry about the confusion for this section -- I don't think we should remove `@@match` per #2 and #4, but override checks for `exec` and `flags`, etc.

File: regexp-diff.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# RegExp changes
+
+```diff
+21.2.3.1 RegExp ( pattern, flags )
+
+The following steps are taken:
+
+-    1. Let patternIsRegExp be ? IsRegExp(pattern).
```
Comment (syg) [inline regexp-diff.md:8]: Pending the discussion in #2 and #4 I'm not sure if this is accurate. What do you think about what we should do for RegExps?

Comment (codehag) [reply regexp-diff.md:8]: I think this has been resolved. I will remove this file.

File: README.md (hunk 3 above)
Comment (codehag) [reply README.md:54]: Don't know if i got this right now...

File: README.md [hunk 7]
```diff
@@ -175,17 +265,37 @@ These changes apply to the following methods.
...
 
+
+### Example
+
+Before this change:
+
+```js
+class R extends RegExp {
+  exec() {
+    return "overriden";
```
Comment (syg) [inline README.md:282]: nit: typo in overridden

File: regexp-diff.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
+# RegExp changes
```
Comment (syg) [inline regexp-diff.md:1]: This file should be removed from the PR now, right?

Comment (codehag) [reply regexp-diff.md]: yep

File: README.md (hunk 1 above)
Comment (codehag) [reply README.md]: should we open an issue around this? cc @syg

**Pull request merged**